tests/cpu/UnitTestPathUtils.h

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#ifndef INCLUDED_OCIO_UNITTEST_PATHUTILS_H
+#define INCLUDED_OCIO_UNITTEST_PATHUTILS_H
+
+#include <iostream>
+#include <sstream>
+#include <map>
+
+#include <OpenColorIO/OpenColorIO.h>
+
+namespace OCIO_NAMESPACE
+{
+
+namespace
+{
+
+//A custom compute hash function for testing
+std::string CustomComputeHash(const std::string &filename)
+{
+    std::ostringstream testHasher;
+    testHasher << std::hash<std::string>{}(filename + "this is custom hash");
+    return testHasher.str();
+}
+
+} //anon.
+
+} // namespace OCIO_NAMESPACE
+
+#endif //INCLUDED_OCIO_UNITTEST_PATHUTILS_H
```

Comment:
The unit test framework is a custom mechanism where we do not add include files. Please move the method definition in the corresponding cpp file (and then delete that new file).

Reply:
I'll surely keep that in mind! I was trying to make a `PathUtilsTester` class for invoking functions from the public API but it didn't work for me so well. Should I be implementing a class like this anyways?

Reply:
If you need an helper class (like the `PathUtilsTester`) then the code must declare & define it in the corresponding test file. For example, the class `PathUtilsTester` include and cpp file content must be moved in PathUtils_tests.cpp.

Note: The class name should be `HashGuard` or `ComputeHashGuard` to better highlight that it only cares of the `ComputeHashFunction`, and it restores the default implementation when the test completes.